services/src/main/java/org/keycloak/protocol/oidc/OIDCClientSecretConfigWrapper.java

```diff
@@ -214,4 +214,203 @@ public String toJson() {
             return "";
         }
     }
+
+    public ReadOnlyRotatedSecretClientModel toRotatedClientModel() {
+        return new ReadOnlyRotatedSecretClientModel();
+    }
+
+    /**
+     * Representation of a client model that passes information from a rotated secret. The goal is to act as a decorator/DTO just providing information and not updating objects persistently.
+     */
+    public class ReadOnlyRotatedSecretClientModel extends AbstractReadOnlyClientStorageAdapter {
```

Comment:
Is it possible to rather inherit from the class `org.keycloak.models.delegate.ClientModelLazyDelegate`? This will help to avoid so much overriden methods. I guess only `getSecret` method would need to be overriden in that case.

Reply:
Perfect, it was exactly the class I was looking for. As the ClientModel interface has several implementations, the one I found closest to what I would like was exactly the one I used. Your suggestion is exactly what I needed. Thanks.